clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1021,45 +1016,52 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
             MPM.addPass(InstrProfiling(*Options, false));
           });
 
-    if (CodeGenOpts.OptimizationLevel == 0) {
-      MPM = PB.buildO0DefaultPipeline(Level, IsLTO || IsThinLTO);
-    } else if (IsThinLTO) {
-      MPM = PB.buildThinLTOPreLinkDefaultPipeline(Level);
-    } else if (IsLTO) {
-      MPM = PB.buildLTOPreLinkDefaultPipeline(Level);
+    if (!CodeGenOpts.DisableSYCLEarlyOpts) {
```

Comment:
I suggest that we reduce the amount of changes we have here by simply adding a one more `if` into the chain, like this:

```
if (CodegenOpts.DisableSYCLEarlyOpts) {
  MPM = PB.buildO0DefaultPipeline(OptimizationLevel::O0, IsLTO || IsThinLTO);
} else if (CodeGenOpts.OptimizationLevel == 0) {
  MPM = PB.buildO0DefaultPipeline(Level, IsLTO || IsThinLTO);
} // the rest of existing code
```